File: python/arrays/sparse_arrays.py
def matchingStrings(strings, queries):
    r = []
    for q in queries:
        m = 0
        for s in strings:
            if s == q:
                m += 1
        r.append(m)
    
    return r



strings = ['aba', 'baba', 'aba', 'xzxb']
queries = ['aba', 'xzxb', 'ab']

print(matchingStrings(strings, queries))

File: php/arrays/sparseArrays.php
<?php

function matchingStrings($strings, $queries){
    $r = [];
    
    foreach( $queries as $q ){
        $m = 0;
        foreach( $strings as $s ){
            if( $s == $q )
                $m += 1;
        }
        $r[] = $m;
    }

    return $r;
}

$strings = ['aba', 'baba', 'aba', 'xzxb'];
$queries = ['aba', 'xzxb', 'ab'];

print_r(matchingStrings($strings, $queries));